File: Python/SQLAlchemy_ORM/bll/services/employee_service.py
from dal.repositories.employee_repository import EmployeeRepository
from dal.data.database import SessionLocal
from dal.entities.employee import Employee

class EmployeeService:
    def __init__(self):
        self.db = SessionLocal()
        self.repo = EmployeeRepository(self.db)

    def get_by_id(self, id: int):
        entity = self.repo.get_by_id(id)
        if not entity:
            raise ValueError(f"Entity {id} not found")
        return entity

    def get_all(self):
        return self.repo.get_all()

    def create(self, entity: Employee) -> Employee:
        return self.repo.create(entity)

    def update(self, id: int, entity: Employee) -> Employee:
        entity = self.repo.update(id, entity)
        if not entity:
            raise ValueError(f"Entity {id} not found for update")
        return entity

    def delete(self, id: int):
        entity = self.repo.delete(id)
        if not entity:
            raise ValueError(f"Entity {id} not found for deletion")
        return entity

File: Node/Sequelize_ORM/src/bll/services/employeeService.js
const EmployeeRepository = require('../../dal/repositories/employeeRepository');

class EmployeeService {
  async getAll() {
    return await EmployeeRepository.getAll();
  }

  async getById(id) {
    return await EmployeeRepository.getById(id);
  }

  async create(data) {
    return await EmployeeRepository.create(data);
  }

  async update(id, data) {
    return await EmployeeRepository.update(id, data);
  }

  async delete(id) {
    return await EmployeeRepository.delete(id);
  }
}

module.exports = new EmployeeService(); 